user_management/api/views.py

```diff
@@ -151,7 +152,8 @@ def initial(self, request, *args, **kwargs):
         try:
             self.user = User.objects.get(pk=uid)
         except User.DoesNotExist:
-            raise Http404()
+            msg = _('Invalid or expired token.')
+            raise PermissionDenied(detail=msg)
```

Comment:
I'm surprised no one's yet mentioned 418 as an option?

400 is the correct code in this case, as none of this code verification is "authentication" - if you are authenticated, or trying to be, you shouldn't be here at all.